sky/server/requests/payloads.py

```diff
@@ -81,9 +81,12 @@ def request_body_env_vars() -> dict:
     env_vars[constants.USER_ENV_VAR] = common_utils.get_current_user_name()
     env_vars[
         usage_constants.USAGE_RUN_ID_ENV_VAR] = usage_lib.messages.usage.run_id
+    if not common.is_api_server_local():
+        # Used in job controller, for local API server, keep the
+        # SKYPILOT_CONFIG env var to use the config for the managed job.
+        env_vars.pop(skypilot_config.ENV_VAR_SKYPILOT_CONFIG, None)
```

Comment:
We need to test the following case:
1. `export SKYPILOT_CONFIG='xxx'; sky api start`
2. In the same terminal, `export SKYPILOT_PROJECT_CONFIG='yyy'; sky launch`. Does this correctly override the configs?
3. In the same terminal, add a .sky.yaml in the CWD, `sky launch`. Does it correctly override the two configs?
4. In a different terminal, no `SKYPILOT_CONFIG` set in env var, does the config is used correctly.

Reply:
`SKYPILOT_PROJECT_CONFIG` will not override `SKYPILOT_CONFIG` on the client request.
We should do 1. in a separate terminal.